### PR TITLE
Clang warnings fix for ART backend

### PR DESCRIPTION
**Pull request by trunkmaster**
These changes are:
- int formatting fixes (vast majority);
- usage of old freetype library (prior to 2.1.8 version). I've not removed `ftfont-old.m` though;
- removed usage of "GNU old-style field designator extension".

### CODE REVIEW

**Comment by trunkmaster** [comment]
Riccardo, do you have systems with old freetype library?

File: Source/art/ftfont.m [hunk 1]
```diff
@@ -22,17 +22,6 @@
    Boston, MA 02110-1301, USA.
 */
 
-#include <ft2build.h>
-#include FT_FREETYPE_H
-
-
-#if (FREETYPE_MAJOR==2) && ((FREETYPE_MINOR<1) || ((FREETYPE_MINOR==1) && (FREETYPE_PATCH<8)))
-
-#import "ftfont-old.m"
```
Comment (fredkiefer) [inline Source/art/ftfont.m]: If we no longer include the file here, we might as well delete it.

File: Source/art/ftfont.m [hunk 2]
```diff
@@ -56,7 +45,7 @@
 
 #import "ftfont.h"
 #import "FTFontEnumerator.h"
-#import "FTFaceInfo.h"
+// #import "FTFaceInfo.h"
```
Comment (fredkiefer) [inline Source/art/ftfont.m]: Please remove this now commented out line.

File: Source/art/ftfont.m [hunk 3]
```diff
@@ -2025,12 +1981,12 @@ static int bezierpath_cubic_to(const FT_Vector *c1, const FT_Vector *c2,
 }
 
 static FT_Outline_Funcs bezierpath_funcs = {
-  move_to:bezierpath_move_to,
-  line_to:bezierpath_line_to,
-  conic_to:bezierpath_conic_to,
-  cubic_to:bezierpath_cubic_to,
-  shift:10,
-  delta:0,
+  .move_to = bezierpath_move_to,
```
Comment (fredkiefer) [inline Source/art/ftfont.m]: This way of assigning values to a struct did not exist when I learned C ages ago. I will have to get used to it :-)

**Comment by rmottola** [comment]
> Riccardo, do you have systems with old freetype library?

How old is "old". I Have only two systems where I still use art, one is current and up to date, the next I can check next week. < 2.1.1 ?

**Comment by trunkmaster** [comment]
On 18 Jan 2020, at 14:17, Riccardo <notifications@github.com> wrote:
> 
> Riccardo, do you have systems with old freetype library?
> 
> How old is "old". I Have only two systems where I still use art, one is current and up to date, the next I can check next week. < 2.1.1 ?
> 
This line from ftfont.m :

#if (FREETYPE_MAJOR==2) && ((FREETYPE_MINOR<1) || ((FREETYPE_MINOR==1) && (FREETYPE_PATCH<8)))

specifies version < 2.1 or < 2.1.8. Could you check if that system can work without art/ftfont-old.m file?

File: Source/art/ftfont.m (hunk 1 above)
Comment (trunkmaster) [reply Source/art/ftfont.m]: I'm waiting for Riccardo results.

File: Source/art/ftfont.m (hunk 2 above)
Comment (trunkmaster) [reply Source/art/ftfont.m]: Done.

**Comment by trunkmaster** [comment]
@rmottola, any news?

**Comment by rmottola** [comment]
@trunkmaster I have some news. First, the old machine (it is the small MIPS minbook which Fred will remember and which for kernel reasons I cannot update to newer versions of Debian) has FreeType 2.2, so in "theory" there should be no issue!
I rebuilt all gnustep "from scratch" natively, to test your patch. It builds, but fails.
I noticed that libart is not netected, so I got a fallback on xlib. I need to investigate that.
Xlib then fails with a crash (the reason I was using libart on this machine in the first place, there are issues with the system fonts. It used not to crash, but to draw no strings on the screen, I think this a "regression", but a different issue.

I will try to report back "why" libart is not being found, I will write to the mailing list, please bear some patience.

**Comment by trunkmaster** [comment]
@rmottola, no problem. Waiting for report on libart issue.

File: Source/art/ftfont.m (hunk 1 above)
Comment (rmottola) [reply Source/art/ftfont.m]: I think the issues I am having are not related with the removal of ftflond-old.m, the freetype release I am using is newer than 2.1.1 on my test machines and the code compiles

**Pull request merged**